stan/math/prim/fun/min.hpp

```diff
@@ -10,6 +10,21 @@
 namespace stan {
 namespace math {
 
+/**
+ * Returns the minimum coefficient of the two specified
+ * scalar arguments.
+ *
+ * @tparam T1 type of first argument (must be arithmetic)
+ * @tparam T2 type of second argument (must be arithmetic)
+ * @param x first argument
+ * @param y second argument
+ * @return minimum value of the two arguments
+ */
+template <typename T1, typename T2, require_all_arithmetic_t<T1, T2>* = nullptr>
+return_type_t<T1, T2> min(T1 x, T2 y) {
```

Comment:
> Will this return an integer if x and y are both integers? 

Yes, it will. That is tested here: https://github.com/stan-dev/math/blob/develop/test/unit/math/mix/meta/return_type_test.cpp#L17